services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -5341,10 +5343,13 @@ private OfflinePushStatusInfo getOfflinePushStatusInfo(
     if (store.isDaVinciPushStatusStoreEnabled() && (versionNumber > 1 || incrementalPushVersion.isPresent())) {
       if (store.getVersion(versionNumber).isPresent()) {
         Version version = store.getVersion(versionNumber).get();
-        Pair<ExecutionStatus, String> daVinciStatusAndDetails =
-            getDaVinciPushStatusAndDetails(version, incrementalPushVersion);
-        ExecutionStatus daVinciStatus = daVinciStatusAndDetails.getFirst();
-        String daVinciDetails = daVinciStatusAndDetails.getSecond();
+        ExecutionStatusWithDetails daVinciStatusAndDetails = PushMonitorUtils.getDaVinciPushStatusAndDetails(
+            pushStatusStoreReader.orElse(null),
+            version.kafkaTopicName(),
+            version.getPartitionCount(),
+            incrementalPushVersion);
+        ExecutionStatus daVinciStatus = daVinciStatusAndDetails.getStatus();
+        String daVinciDetails = daVinciStatusAndDetails.getDetails();
```

Comment:
I guess querying DaVinci status logic won't be required any more since `PushStatusCollector` will consolidate DaVinciStatus and Server Status and update `OfflinePush` znode, right?

Reply:
I guess what you mean here is we can try to fetch from collector's current status? Probably yeah...need some refactoring though.

Reply:
Added...but I somehow feel like the logic between Admin/Monitor class(3 layers of basically one thing)/ new Collector is becoming more complicated... I don't know if that's also your feeling or not but it would be nice if we do the real refactor in some other PR to clean it up.

Reply:
No, I was thinking querying `OfflinePushStatus` is good enough since the aggregated status on top of DaVinci/Server will be persisted in `OfflinePushStatus` eventually.

Reply:
But actually here for incremental push it is not the case right? It is not monitored there if I understand it correctly.

Reply:
And I am not sure my latest implementation of querying collector is good, I feel like it is making the logic more complicated.

Reply:
I see.
For incremental push, you are right.
For batch-only push, querying offline push status should be good enough.
I guess incremental push should use a different code path from batch push since it doesn't rely on the `offline push status` at all.